README.md

```diff
@@ -17,34 +17,63 @@ Docker versions.
+  - `docker_version`: this variable controls the version of Docker that is installed. Required.
     If version `1.5.0` is selected, LXC Docker will be used; otherwise the stated version of
     Docker Engine will be installed (if available).
-  - `docker_daemon_flags` : Empty by default. This variable holds flags that will be passed to
+  - `docker_daemon_flags`: Empty by default. This variable holds flags that will be passed to
     the Docker daemon on startup. (This is implemented by modifying the file `/etc/default/docker`.)
-  - `cgroup_lite_pkg_state` : When installing on an Ubuntu 13.10 host, the role will install the
-    `cgroup-lite` package to provide the required cgroups support. This variable can be set to
-    `latest` - the default - or to `present`. In the former case, the package will be updated, if
-    necessary, when the role is run. In the latter, the package will only be added if it is not
-    present.
-  - `kernel_pkg_state` : For 13.04+, this role will install a `linux-image-extra-<version>`
-    package. This parameter works the same way as `cgroup_lite_package_state`, except controlling
-    this package.
+  - `docker_daemon_startup_retries`: this variable controls how many times we poll docker to
+    confirm it is running after we start or restart it before giving up. Defaults to 10.
+
+Documentation
+-------------
+
+The documentation for working with Docker on Ubuntu is available online but there has been
+some refactoring of the documentation since the original writing.
+
+  * https://docs.docker.com/engine/installation/linux/ubuntu/
+  * (old) https://github.com/docker/docker.github.io/blob/master/engine/installation/linux/ubuntulinux.md
+  * (new) https://github.com/docker/docker.github.io/blob/master/engine/installation/linux/ubuntu.md
+
+The *old* documentation is what was previously available online and the *new* documentation is what is
+currently available online. As of this writing the documentation is at commit '45a19ec' & '9093e0a' respectively.
+
+  * (old) https://github.com/docker/docker.github.io/blob/45a19ec/engine/installation/linux/ubuntulinux.md
+  * (new) https://github.com/docker/docker.github.io/blob/9093e0a/engine/installation/linux/ubuntu.md
+
+Links to documentation will therefore get pinned to a particular commit to maintain access
+to historical information which may get removed (or moved) in later versions. Maintainers
+should check the master branch when updating the role and update links when possible.
+
+Currently the new documentations says docker is only supported on 14.04[LTS], 16.04[LTS], & 16.10
+but the old documentation has some instructions for 12.04[LTS].
+
+12.04 may not have support due to issues with older kernels. Docker documentation mentions some
+prerequisites when installing from a binary that could be informative and there is a known
+issue with docker running on linux kernels less than 3.19 that could imply 12.04 isn't supported.
```

Comment:
Presumably we're running docker on 12.04 in various places?

If so, do we have an answer to this? Really we should get off of 12.04 (and not just b/c of docker), but we have a terrible track record with things like this. Do we suspect that 12.04 plus an updated kernel is good enough? Has your new docker_upgrade ansible role been tested on 12.04 ?